### ISSUE TITLE
Single map inside a list, list is not preserved

### ISSUE THREAD

**Issue by jpaasen** [fangq/jsonlab#15]
I have the following simple json-file that does not remain the same if you read it using loadjson and then store it with savejson (using the master branch and Matlab 2014b).

```json
{
	"test1":[
		{
			"A":0,
			"B":1
		}
	],
	"test2":{
		"C":2,
		"D":3
	}
}
```

The '[]' brackets are removed. I know that arrays with single elements is quite problematic in Matlab, but it looks like you now, compared to older versions, are able to create a cell-array holding a single struct. At least that is what I see printed in the console after loading the example file. 

```
test = loadjson('test.json')
test = 
     test1 : {[1x1 struct]}
     test2 : [1x1 struct]
```

Would it be possible to use this information when storing the json file to preserve the single element list?

**Comment by cyrilpic** [comment]
:+1: 
Having the same issue. An array with 1 element is not the same as the element directly.

Changing line 188 and 219 (replacing `len > 1` by `len > 0`) does fix this in my case, but I am not well aware of possible side effects.

**Comment by jpaasen** [comment]
Thanks. It works for me as well.

Any chance to have the following file preserved after load-and-store?
```json
{
    "test1":[0],
    "test2":1
}
```

**Comment by fangq** [comment]
Closed as a duplicate of issue #8. Please visit issue #8 for further discussions.
